var/spack/repos/builtin/packages/hip/package.py

```diff
@@ -120,15 +99,24 @@ def setup_dependent_build_environment(self, env, dependent_spec):
         env.set('HIP_CLANG_PATH', rocm_prefixes['llvm-amdgpu'].bin)
         env.set('HSA_PATH', rocm_prefixes['hsa-rocr-dev'])
         env.set('ROCMINFO_PATH', rocm_prefixes['rocminfo'])
-        env.set('DEVICE_LIB_PATH', rocm_prefixes['rocm-device-libs'].lib)
+        env.set('DEVICE_LIB_PATH', rocm_prefixes['device_lib_path'])
         env.set('HIP_PATH', rocm_prefixes['rocm-path'])
         env.set('HIPCC_COMPILE_FLAGS_APPEND',
-                '--rocm-path={0}'.format(rocm_prefixes['rocm-path']))
+                '--rocm-path={0}'.format(rocm_prefixes['device_lib_path']))
+
+    def setup_run_environment(self, env):
+        self.set_variables(env)
+
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        self.set_variables(env)
 
         if 'amdgpu_target' in dependent_spec.variants:
             arch = dependent_spec.variants['amdgpu_target'].value
             if arch != 'none':
```

Comment:
Didn't see this comment until now, but yeah, it should be fine, because any_combination_of allows for an empty collection, which is defined to be the string `'none'`.